### DISCUSSION TITLE
(node:76700) UnhandledPromiseRejectionWarning: Error: Invalid request.

### DISCUSSION THREAD

**Discussion by sannyii** [iTwin/itwinjs-core#3463] [Q&A]
Hi, I got a error when I put iTwin on Tecent Cloud.

`(node:76700) UnhandledPromiseRejectionWarning: Error: Invalid request.
    at Function.parseRequest (E:\wangxiao\CODE\itwin-example-for-china\node_modules\@bentley\imodeljs-common\lib\rpc\web\WebAppRpcRequest.js:70:19)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async protocol.handleOperationPostRequest (E:\wangxiao\CODE\itwin-example-for-china\node_modules\@bentley\imodeljs-common\lib\rpc\web\WebAppRpcProtocol.js:83:25)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:76700) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 373)
(node:76700) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

This  error occured on this file's this line    
file is     **WeBappRpcRequest.js**
code is  **if (!request.id) {
            throw new IModelError_1.IModelError(bentleyjs_core_1.BentleyStatus.ERROR, `Invalid request.`);
        }**

`/** Parses a request. */
    static async parseRequest(protocol, req) {
        const operation = protocol.getOperationFromPath(req.url);
        const parsedHeaders = this.parseHeaders(protocol, req);
        const request = Object.assign(Object.assign({}, parsedHeaders), { operation: {
                interfaceDefinition: operation.interfaceDefinition,
                operationName: operation.operationName,
                interfaceVersion: operation.interfaceVersion,
            }, method: req.method, path: req.url, parameters: operation.encodedRequest ? WebAppRpcRequest.parseFromPath(operation) : await WebAppRpcRequest.parseFromBody(req), caching: operation.encodedRequest ? RpcConstants_1.RpcResponseCacheControl.Immutable : RpcConstants_1.RpcResponseCacheControl.None });
        request.ip = req.ip;
        request.protocolVersion = 0;
        if (protocol.protocolVersionHeaderName) {
            const version = req.header(protocol.protocolVersionHeaderName);
            if (version) {
                request.protocolVersion = parseInt(version, 10);
            }
        }
        if (!request.id) {
            throw new IModelError_1.IModelError(bentleyjs_core_1.BentleyStatus.ERROR, `Invalid request.`);
        }
        return request;
    }`

please tell me how can  I solve it ?
Thank you ~

**Comment by swbsi** [comment]
We pass the request ID using the "X-Correlation-Id" header.
Is it possible that this header is blocked by any middleware or proxy layers within Tencent?
